pyproject.toml

```diff
@@ -75,6 +76,11 @@ docs = [
     "myst-parser",
     "sphinx-autobuild",
 ]
+# Mark these as extras so that installs pass on python 3.11
+devextras = [
+    "pyarrow",
+    "viztracer",
+]
```

Comment:
@laurentS wdyt of splitting these between two extras:

```
samples = ["pyarrow"]
debugging = ["viztracer"]
```

We may want to [drop viztracer](https://github.com/meltano/sdk/issues/1124) as you suggest but it's probably not worth increasing the scope of this PR.

Reply:
done in 740f547